File: algorithms/easy/longest-common-prefix.py
#https://leetcode.com/problems/longest-common-prefix/

class Solution:
    def longestCommonPrefix(self, strs: list[str]) -> str:
        longest = max(strs, key=len)
        for i, char in enumerate(longest):
            for j in strs:
                if j[i] != char:
                    return longest[:i]
        return longest

solution = Solution().longestCommonPrefix(["flower","flow","flight"])
print(solution)

# !TODO

File: cpp/longest_common_prefix.cpp
#include <iostream>
#include <vector>
using namespace std;

class LongestCommonPrefix{
public:
    static string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        string prefix = strs[0];

        for(const string& s : strs){
            while (s.find(prefix) != 0){
                prefix = prefix.substr(0, prefix.length() - 1);
            }
        }
        return prefix;
    }
};